django/db/backends/mysql/schema.py

```diff
@@ -69,6 +69,10 @@ def _field_should_be_indexed(self, model, field):
                 field.get_internal_type() == 'ForeignKey' and
                 field.db_constraint):
             return False
+        # MySQL's blob/text/json fields do not support indexing of whole field
+        # https://code.djangoproject.com/ticket/27859
+        if field.get_internal_type() == 'TextField':
```

Comment:
Is this the correct comparison? Looks a bit different from the condition in `skip_default()` which seems to be skipping similar field types.